src/main/java/uk/gov/ch/oauth/Oauth2.java

```diff
@@ -111,11 +111,12 @@ public boolean oauth2VerifyNonce(final String nonce) {
      *
      * @return The Nonce String from within the session or null if not found.
      */
-    private String getSessionNonce() {
+    public String getSessionNonce(final Session chSession) {
         String oauth2Nonce = null;
         try {
-            final Map<String, Object> data = sessionService.getSessionDataFromContext();
-            oauth2Nonce = (String) data.getOrDefault(SessionKeys.NONCE.getKey(), null);
+            final Map<String, Object> data = chSession.getData();
+            oauth2Nonce = (String) data.remove(SessionKeys.NONCE.getKey());
```

Comment:
If we're removing the nonce, then we must make sure we're accounting for the possibility of a potential null pointer upstream. I would feel more comfortable updating the value to a blank which is a direct copy of the perl implementation, however as long as we have thought about a potential null pointer then I suppose it's fine.